src/main/java/com/couchbase/lite/support/MultipartReader.java

```diff
@@ -69,10 +68,9 @@ private boolean memcmp(byte[] array1, byte[] array2, int len) {
     }
 
     public Range searchFor(byte[] pattern, int start) {
-
         KMPMatch searcher = new KMPMatch();
-        int matchIndex = searcher.indexOf(buffer.toByteArray(), pattern, start);
-
+        int buffLen = Math.min(buffer.length(), buffer.buffer().length);
```

Comment:
I understood that setLength() value is expected to be less than the current capacity so buffer.length is always less than or equal to buffer.buffer().length be default but I might understand it wrong. This is not a big deal though.

```
**
     * Sets the length of the buffer. The new length value is expected to be
     * less than the current capacity and greater than or equal to
     * <code>0</code>.
     *
     * @param      len   the new length
     * @throws     IndexOutOfBoundsException  if the
     *               <code>len</code> argument is greater than the current
     *               capacity of the buffer or less than <code>0</code>.
     */
    public void setLength(int len) {
        if (len < 0 || len > this.buffer.length) {
            throw new IndexOutOfBoundsException("len: "+len+" < 0 or > buffer len: "+this.buffer.length);
        }
        this.len = len;
    }
```